### PR TITLE
Photometry

### PR DESCRIPTION
**Pull request by JaerongA**

### CODE REVIEW

File: workflow/pipeline/core.py [hunk 1]
```diff
@@ -27,15 +27,15 @@
 session.activate(db_prefix + 'session', linking_module=__name__)
 
 
-# Declare SkullReference table for use in element-array-ephys
+# Declare SurgeryCoordinateReference table for use in element-array-ephys
 @lab.schema
-class SkullReference(dj.Lookup):
+class SurgeryCoordinateReference(dj.Lookup):
```
Comment (ttngu207) [inline workflow/pipeline/core.py]: Can we name this `BrainCoordinateReference`?

Comment (JaerongA) [reply workflow/pipeline/core.py]: ok, I'll make that change.

File: workflow/pipeline/ephys.py [hunk 2]
```diff
@@ -3,20 +3,22 @@
...
+)
 
-__all__ = ['ephys', 'probe']
+__all__ = ["ephys", "probe"]
 
 
 # ------------- Activate "ephys" schema -------------
 
-SkullReference = lab.SkullReference
+SurgeryCoordinateReference = lab.SurgeryCoordinateReference
```
Comment (ttngu207) [inline workflow/pipeline/ephys.py]:
```suggestion
SkullReference = lab.SurgeryCoordinateReference
```
The `element-array=ephys` expects a table named `SkullReference` to activate

File: workflow/pipeline/photometry.py [hunk 3]
```diff
@@ -1,104 +1,324 @@
...
+    """
+
+
+@schema
+class FiberPhotometry(dj.Imported):
+    definition = """
+    -> session.Session
+    fiber_id            : tinyint unsigned    
+    ---
+    side                : enum("left", "right")
```
Comment (ttngu207) [inline workflow/pipeline/photometry.py]:
```suggestion
    -> lab.Hemisphere
```

File: workflow/pipeline/photometry.py [hunk 4]
```diff
@@ -1,104 +1,324 @@
...
+    def make(self, key):
+
+        import pandas as pd
+        import numpy as np
+        from pathlib import Path
+        import tomli
+        import workflow.utils.photometry_preprocessing as pp
+        from workflow.utils import demodulation
+        import tdt
+        from copy import deepcopy
```
Comment (ttngu207) [inline workflow/pipeline/photometry.py]: some duplicated imports here

File: workflow/pipeline/photometry.py [hunk 5]
```diff
@@ -1,104 +1,324 @@
...
+                        "emission_color": emission_color,
+                        "sensor_protein_name": sensor_protein,
+                        "excitation_wavelength": emission_wavelength,
+                        "trace": downsampled_states_df[trace_name].values,
+                    }
+                )
+
+        self.insert(fiber_photometry)
+        self.Trace.insert(trace_list)
+        self.TimeOffset.insert1([*key.values(), time_offset])
```
Comment (ttngu207) [inline workflow/pipeline/photometry.py]:
```suggestion
        self.TimeOffset.insert1({**key, "time_offset": time_offset})

```

File: workflow/pipeline/core.py [hunk 6]
```diff
@@ -1,60 +1,188 @@
...
+    """
+
+
+lab.BrainRegion = BrainRegion
+
+
+@lab.schema
+class Hemisphere(dj.Lookup):
+    definition = """
+    hemisphere: varchar(32)
```
Comment (ttngu207) [inline workflow/pipeline/core.py]: Is `varchar(32)` too much here? Can we get away with 16 or 8?

File: workflow/pipeline/core.py [hunk 7]
```diff
@@ -1,60 +1,188 @@
...
+    class Coordinate(dj.Part):
+        definition = """
+        -> master
+        ---
+        ap            : decimal(3, 3)           # (um) anterior-posterior; ref is 0
+        -> lab.BrainCoordinateReference.proj(ap_ref='reference')
+        ml            : decimal(3, 3)           # (um) medial axis; ref is 0 
+        -> lab.BrainCoordinateReference.proj(ml_ref='reference')
+        dv            : decimal(3, 3)           # (um) dorso-ventral axis; ref is 0; more ventral is more negative
+        -> lab.BrainCoordinateReference.proj(dv_ref='reference')
```
Comment (ttngu207) [inline workflow/pipeline/core.py]: In mysql, `decimal(x, y)` means a total of `x` digit with `y` number of decimal points.
`decimal(3,3)` means the largest value you can have is `.999`.
I think you mean to use `decimal(6, 3)`

File: workflow/pipeline/ephys.py [hunk 8]
```diff
@@ -3,20 +3,22 @@
...
+)
 
-__all__ = ['ephys', 'probe']
+__all__ = ["ephys", "probe"]
 
 
 # ------------- Activate "ephys" schema -------------
 
-SkullReference = lab.SkullReference
+BrainCoordinateReference = lab.BrainCoordinateReference
```
Comment (ttngu207) [inline workflow/pipeline/ephys.py]:
```suggestion
SkullReference = lab.BrainCoordinateReference
```
The `element-array-ephys` expect a table named `SkullReference` in order to be activated

File: workflow/pipeline/imaging.py [hunk 9]
```diff
@@ -16,16 +16,4 @@
...
-@lab.schema
-class BrainRegion(dj.Manual):
-    definition = """
-    acronym: varchar(32)
-    ---
-    region_name: varchar(128)
-    """
-
-
-Location = BrainRegion
```
Comment (ttngu207) [inline workflow/pipeline/imaging.py:29]:
```
Location = lab.BrainRegion
```
The `element-calcium-imaging` needs a table named `Location` in order to be activated

**Review by ttngu207 — CHANGES_REQUESTED** [review]

File: workflow/pipeline/core.py (hunk 7 above)
Comment (JaerongA) [reply workflow/pipeline/core.py]: yes, forgot to fix it there.

**Review by ttngu207 — APPROVED** [review]

**Pull request merged**